### PR TITLE
[Merged by Bors] - Add TextureAtlas stress test based on many_sprites and sprite_sheet examples

### LINKED ISSUE #5073: Add texture atlas stress test
## What problem does this solve or what need does it fill?

While we can test standard 2D performance in realistic ways, we have no comparable tests that involve texture atlases.

This is particularly important as we start to build out 2D animation features.

## What solution would you like?

Add a [stress test](https://github.com/bevyengine/bevy/tree/main/examples/stress_tests) that uses [`TextureAtlas`](https://dev-docs.bevyengine.org/bevy/sprite/struct.TextureAtlas.html) heavily.

This would likely be most realistic if it involved many desynchronized sprite-sheet animations (which currently must be manually operated on). A naive tile-map texture atlas stress test may also be useful.

## Additional context

This is required by #5072.

### PR DESCRIPTION
**Pull request by james-j-obrien**
# Objective

Intended to close #5073

## Solution

Adds a stress test that use TextureAtlas based on the existing many_sprites test using the animated sprite implementation from the sprite_sheet example.

In order to satisfy the goals described in #5073 the animations are all slightly offset.

Of note is that the original stress test was designed to test fullstrum culling. I kept this test similar as to facilitate easy comparisons between the use of TextureAtlas and without.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
@ManevilleF, here's a nice little stress test for the changes in #5072 :)

File: examples/stress_tests/many_animated_sprites.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+// System for printing the number of sprites on every tick of the timer
+fn print_sprite_count(time: Res<Time>, mut timer: Local<PrintingTimer>, sprites: Query<&Sprite>) {
+    timer.tick(time.delta());
+
+    if timer.just_finished() {
+        info!("Sprites: {}", sprites.iter().count(),);
+    }
+}
```
Comment (mockersf) [inline examples/stress_tests/many_animated_sprites.rs:132]: This count is not right, it always displays 0

**Review by mockersf — CHANGES_REQUESTED** [review]

File: examples/stress_tests/many_animated_sprites.rs (hunk 1 above)
Comment (james-j-obrien) [reply examples/stress_tests/many_animated_sprites.rs]: Should be fixed now, just a change from Sprite to TextureAtlasSprite

**Review by ManevilleF — APPROVED** [review]

**Pull request closed without merging**